pkg/server/server.go

```diff
@@ -222,15 +242,90 @@ func (s *Server) Run(ctx context.Context) error {
 		KeyFile:       s.cfg.EtcdClientInfo.KeyFile,
 		TrustedCAFile: s.cfg.EtcdClientInfo.TrustedCAFile,
 	}
+
+	// TODO(ncdc): I thought I was going to need this, but it turns out this breaks the CRD controllers because they
+	// try to issue Update() calls using the * client, which ends up with the cluster name being set to the default
+	// admin cluster in handler.go. This means the update calls are likely going against the wrong logical cluster.
+	// serverOptions.APIExtensionsNewClientFunc = func(config *rest.Config) (apiextensionsclient.Interface, error) {
+	// 	const crossCluster = "*"
+	// 	clusterClient, err := apiextensionsclient.NewClusterForConfig(config)
+	// 	if err != nil {
+	// 		return nil, err
+	// 	}
+	// 	client := clusterClient.Cluster(crossCluster)
+	// 	return client, nil
+	// }
+
+	loopbackClientCert, loopbackClientCertKey, err := serverOptions.SecureServing.NewLoopbackClientCert()
+	if err != nil {
+		return err
+	}
+	serverOptions.SecureServing.LoopbackCertPEM = loopbackClientCert
+	serverOptions.SecureServing.LoopbackKeyPEM = loopbackClientCertKey
+	serverOptions.SecureServing.LoopbackToken = uuid.New().String()
+	loopbackClientConfig, err := serverOptions.SecureServing.NewLoopbackClientConfig(serverOptions.SecureServing.LoopbackToken, serverOptions.SecureServing.LoopbackCertPEM)
+	if err != nil {
+		return err
+	}
+
+	// FIXME: switch to a single set of shared informers
```

Comment:
We already have one of these for the workspace controller - can we at least use the same factory for cross-cluster bits?

Reply:
I plan on unifying factories in a follow-up PR 👍